core/src/main/java/io/questdb/cairo/map/FastMapRecord.java

```diff
@@ -157,17 +161,34 @@ private FastMapRecord(
             Long256Impl[] keyLong256B
     ) {
         this.keySize = keySize;
+        this.valueSize = valueSize;
         this.valueOffsets = valueOffsets;
         this.keyTypes = keyTypes;
         this.splitIndex = splitIndex;
-        this.value = new FastMapValue(valueOffsets);
+        this.value = new FastMapValue(valueSize, valueOffsets);
         this.csA = csA;
         this.csB = csB;
         this.bs = bs;
         this.keyLong256A = keyLong256A;
         this.keyLong256B = keyLong256B;
     }
 
+    @Override
+    public void copyToKey(MapKey destKey) {
+        FastMap.BaseKey destFastKey = (FastMap.BaseKey) destKey;
+        int keySize = this.keySize;
+        if (keySize == -1) {
+            keySize = Unsafe.getUnsafe().getInt(startAddress);
+        }
+        destFastKey.copyFromRawKey(keyAddress, keySize);
```

Comment:
I don't see a check that `keyAddress` is allocated to have `keySize` copied in, where does it happen?

Reply:
`keyAddress` is set on `FastMapRecord#of()` call, so on record initialization. We only call `copyFromRawKey` on an initialized record, so there are no extra assertions here.

Reply:
Sorry, why `destFastKey` is allocated enough to accommodate the copy?

Reply:
Got it. There is a `checkSize` call in `FastMap.VarSizeKey#copyFromRawKey()` implementation, so the memory is guaranteed to be pre-allocated.

Reply:
`FastMap.FixedSizeKey` is also pre-allocated, but this happens on `FastMap#withKey()` call.